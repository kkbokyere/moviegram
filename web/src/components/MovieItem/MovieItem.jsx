import React from 'react';
import PropTypes from 'prop-types';

const MovieItem = ({ img, title, showTimes = [], id }) => {
    return (
        <article className="movie-item">
            <div>
            <img className="movie-item__img" src={img} alt={title}/>
            </div>
            <div className="movie-item__wrapper">
                <h1 className="movie-item__title">{title}</h1>
                <div className="movie-item__showtimes">
                    <h3>Show Times:</h3>
                    {showTimes.map(({time, cinema}, index) => <span className="movie-item__time" key={index}>{time} @ {cinema}</span>)}
                </div>
                <button className="btn btn-primary">See more</button>
            </div>
        </article>
    );
};

MovieItem.propTypes = {
    img: PropTypes.string,
    title: PropTypes.string,
    showTimes: PropTypes.array,
    id: PropTypes.string,
};

export default MovieItem;