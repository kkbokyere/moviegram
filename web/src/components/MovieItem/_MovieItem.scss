.movie-item {
  border-bottom: 1px solid $primary-grey;
  display: flex;
  margin-bottom: $default-padding;
  padding: $default-padding 0;
}

.movie-item__title {
}

.movie-item__img {
  margin-right: $default-padding;
}

.movie-item__showtimes {

  margin-bottom: $default-padding;
}

.movie-item__time {
  display: block;
  margin-bottom: 10px;
}