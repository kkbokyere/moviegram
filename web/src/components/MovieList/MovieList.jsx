import React from 'react';
import MovieItem from '../MovieItem';

const MovieList = ({data}) => {
    return (
        <section className="movie-list">
            {data.map((movieData) => <MovieItem key={movieData.id} {...movieData}/>)}
        </section>
    );
};

export default MovieList;