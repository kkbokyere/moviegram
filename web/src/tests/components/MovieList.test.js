import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import MovieList from '../../components/MovieList';
import MovieItem from "../../components/MovieItem";

describe('MovieList Component', () => {
    it('should render MovieList', () => {
        const wrapper = renderer.create(<MovieList data={[{}]} />).toJSON();

        expect(wrapper).toMatchSnapshot()
    });

    it('should render MovieList with movie Items', () => {
        const moviesArray = [{
            "id": "1",
            "title": "AVENGERS: ENDGAME",
        }];
        const wrapper = mount(<MovieList data={moviesArray} />);

        expect(wrapper.find(MovieItem).length).toEqual(moviesArray.length)
    });
});