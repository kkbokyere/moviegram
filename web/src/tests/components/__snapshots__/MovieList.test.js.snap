// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList Component should render MovieList 1`] = `
<section
  className="movie-list"
>
  <article
    className="movie-item"
  >
    <div
      className="movie-item__img"
    >
      <picture>
        <source
          media="(min-width: 768px)"
        />
        <img />
      </picture>
    </div>
    <div
      className="movie-item__wrapper"
    >
      <h1
        className="movie-item__title"
      />
      <div
        className="movie-item__showtimes"
      >
        <h3>
          Show Times:
        </h3>
      </div>
      <button
        className="btn btn-primary"
      >
        See more
      </button>
    </div>
  </article>
</section>
`;

exports[`MovieList Component should render MovieList 2`] = `
<section
  className="movie-list"
>
  <article
    className="movie-item"
  >
    <div
      className="movie-item__img"
    >
      <picture>
        <source
          media="(min-width: 768px)"
        />
        <img />
      </picture>
    </div>
    <div
      className="movie-item__wrapper"
    >
      <h1
        className="movie-item__title"
      />
      <div
        className="movie-item__showtimes"
      >
        <h3>
          Show Times:
        </h3>
      </div>
      <button
        className="btn btn-primary"
      >
        See more
      </button>
    </div>
  </article>
</section>
`;

exports[`MovieList Component should render MovieList with movie Items 1`] = `
<section
  className="movie-list"
>
  <article
    className="movie-item"
  >
    <div
      className="movie-item__img"
    >
      <picture>
        <source
          media="(min-width: 768px)"
        />
        <img
          alt="AVENGERS: ENDGAME"
        />
      </picture>
    </div>
    <div
      className="movie-item__wrapper"
    >
      <h1
        className="movie-item__title"
      >
        AVENGERS: ENDGAME
      </h1>
      <div
        className="movie-item__showtimes"
      >
        <h3>
          Show Times:
        </h3>
      </div>
      <button
        className="btn btn-primary"
      >
        See more
      </button>
    </div>
  </article>
</section>
`;

exports[`should render MovieItem 1`] = `
<article
  className="movie-item"
>
  <div
    className="movie-item__img"
  >
    <picture>
      <source
        media="(min-width: 768px)"
      />
      <img />
    </picture>
  </div>
  <div
    className="movie-item__wrapper"
  >
    <h1
      className="movie-item__title"
    />
    <div
      className="movie-item__showtimes"
    >
      <h3>
        Show Times:
      </h3>
    </div>
    <button
      className="btn btn-primary"
    >
      See more
    </button>
  </div>
</article>
`;

exports[`should render MovieList 1`] = `
<section
  className="movie-list"
>
  <article
    className="movie-item"
  >
    <div
      className="movie-item__img"
    >
      <picture>
        <source
          media="(min-width: 768px)"
        />
        <img />
      </picture>
    </div>
    <div
      className="movie-item__wrapper"
    >
      <h1
        className="movie-item__title"
      />
      <div
        className="movie-item__showtimes"
      >
        <h3>
          Show Times:
        </h3>
      </div>
      <button
        className="btn btn-primary"
      >
        See more
      </button>
    </div>
  </article>
</section>
`;
